src/server/game/Movement/MovementGenerators/TargetedMovementGenerator.cpp

```diff
@@ -33,7 +33,7 @@ void TargetedMovementGeneratorMedium<T, D>::_setTargetLocation(T* owner, bool up
     if (!i_target.isValid() || !i_target->IsInWorld())
         return;
 
-    if (owner->HasUnitState(UNIT_STATE_NOT_MOVE))
+    if (owner->HasUnitState(UNIT_STATE_NOT_MOVE | UNIT_STATE_CASTING))
```

Comment:
There isnt even an SPELL_ATTR_EX5_CAN_CHANNEL_WHEN_MOVING flag on trinity, neither is there a IsAbleMoveWhenCast function. so yea "works like a charm" but not on stock TC